host networking pods should be deployed with dnsPolicy set

Unless there's a specific reason not to, host networking pods should have their dnsPolicy set to `dnsPolicy: ClusterFirstWithHostNet` to allow the executables in the pods to leverage kube-dns. 

The triggering example scenario for this issue is enabling Jaeger tracing in the cmd-forwarder-* pods.  Found during https://github.com/networkservicemesh/cmd-forwarder-vpp/issues/344

Kubernetes docs are pretty explicit on using `ClusterFirstWithHostNet` for host networking pods:
https://kubernetes.io/docs/concepts/services-networking/dns-pod-service/#pod-s-dns-policy

To repro, you can test with the patch for networkservicemesh/cmd-forwarder-vpp#344 and the info in https://github.com/tiswanso/deployments-k8s/tree/fix-jaegerenv/examples/features/jaeger

The forwarder pod will throw the following error:
```
2021/09/23 14:53:18 [ERROR] ERROR: cannot init Jaeger: lookup simplest-agent.observability on <-node's default DNS server (redacted)->:53: no such host
```

Setting the dnsPolicy to `ClusterFirstWithHostNet`, fixes it.

As to why `registry` lookup works, I don't have time to look into it but I'm guessing the nsmgr (or lib) code defaults to use localhost when lookup fails.

--- a/apps/forwarder-sriov/forwarder.yaml
+++ b/apps/forwarder-sriov/forwarder.yaml
@@ -15,6 +15,7 @@ spec:
         app: forwarder-sriov
     spec:
       hostNetwork: true
+      dnsPolicy: ClusterFirstWithHostNet
       containers:
         - image: ghcr.io/networkservicemesh/ci/cmd-forwarder-sriov:2586404
           imagePullPolicy: IfNotPresent

--- a/apps/forwarder-vpp/forwarder.yaml
+++ b/apps/forwarder-vpp/forwarder.yaml
@@ -15,6 +15,7 @@ spec:
         app: forwarder-vpp
     spec:
       hostNetwork: true
+      dnsPolicy: ClusterFirstWithHostNet
       containers:
         - image: ghcr.io/networkservicemesh/ci/cmd-forwarder-vpp:5cd6707
           imagePullPolicy: IfNotPresent

